### PR TITLE
Clean up gulp role

### PR DESCRIPTION
**Pull request by sephii**
Clean up the `gulp` role and `nodejs` role to put the creation of the `package.json` file in the `nodejs` role instead of the `gulp` role. The `gulp` role now only patches the existing `package.json` file. Also I think this would close https://github.com/liip/drifter/issues/161.

- [x] Documentation is written
- [x] `parameters.yml.dist` is updated
- [x] `playbook.yml.dist` is updated
- [x] [Changelog](https://github.com/liip/drifter/blob/master/CHANGELOG.md) was updated

### CODE REVIEW

File: docs/roles/others.rst [hunk 1]
```diff
@@ -24,6 +24,11 @@ Parameters
 -  **nodejs\_distro** : Is automatically set to either 'jessie' or
    'wheezy' based on available information, you can also put an Ubuntu
    codename here.
+-  **nodejs_package_json_path**: where should the package.json file be
+   created, defaults to ``<gulp_directory>/package.json``
```
Comment (LeBenLeBen) [inline docs/roles/others.rst]: Feels weird to mention `gulp_directory` in the Node role, is that on purpose?

**Comment by sephii** [comment]
@LeBenLeBen I updated the PR with the requested changes, could you please check again?

File: docs/roles/others.rst [hunk 2]
```diff
@@ -24,6 +24,15 @@ Parameters
 -  **nodejs\_distro** : Is automatically set to either 'jessie' or
    'wheezy' based on available information, you can also put an Ubuntu
    codename here.
+-  **nodejs_package_json_template**: template to use for the creation of the initial ``package.json`` file. Defaults to
+  ``package.json.j2``, or ``package.json.gulp.j2`` if you're using the gulp role. See the
+  ``provisioning/roles/nodejs/templates`` directory for the list of available templates.
+   created, defaults to ``<root_directory>/package.json``
```
Comment (LeBenLeBen) [inline docs/roles/others.rst]: I think you forgot to drop the line 30.

File: provisioning/roles/nodejs/tasks/main.yml [hunk 3]
```diff
@@ -32,3 +32,12 @@
...
   become: yes
   when: nodejs_with_yarn
+
+- name: Create default package.json
+  template: src={{ nodejs_package_json_template }} dest={{ nodejs_package_json_path }} force=no
+
+- name: Install package.json packages
+  command: /bin/true
+  notify:
+    - install package.json
```
Comment (LeBenLeBen) [inline provisioning/roles/nodejs/tasks/main.yml]: Something like "Installing npm packages" would be better I think.

**Comment by sephii** [comment]
@LeBenLeBen I fixed what you asked, can you please check again?

File: provisioning/roles/nodejs/templates/package.json.j2 [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "{{ project_name }}",
+  "version": "1.0.0",
+  "author": "{{ nodejs_package_json_author }}",
+  "license": "private",
+  "private": true,
+  "devDependencies": {
```
Comment (LeBenLeBen) [inline provisioning/roles/nodejs/templates/package.json.j2]: We should probably not include any dependencies in the default package.json

**Pull request merged**